cluster-autoscaler/simulator/scheduling/hinting_simulator.go

```diff
@@ -85,40 +80,45 @@ func (s *HintingSimulator) TrySchedulePods(clusterSnapshot clustersnapshot.Clust
 	return statuses, similarPods.OverflowingControllerCount(), nil
 }
 
-func (s *HintingSimulator) findNodeWithHints(clusterSnapshot clustersnapshot.ClusterSnapshot, pod *apiv1.Pod, isNodeAcceptable func(*framework.NodeInfo) bool) (string, error) {
+func (s *HintingSimulator) tryScheduleUsingHints(clusterSnapshot clustersnapshot.ClusterSnapshot, pod *apiv1.Pod, isNodeAcceptable func(*framework.NodeInfo) bool) (string, error) {
 	hk := HintKeyFromPod(pod)
 	if hintedNode, hasHint := s.hints.Get(hk); hasHint {
-		if err := s.predicateChecker.CheckPredicates(clusterSnapshot, pod, hintedNode); err == nil {
+		nodeInfo, err := clusterSnapshot.GetNodeInfo(hintedNode)
+		if err != nil {
+			return "", err
+		}
+		if !isNodeAcceptable(nodeInfo) {
+			return "", nil
+		}
+		if err := clusterSnapshot.SchedulePod(pod, hintedNode); err == nil {
 			s.hints.Set(hk, hintedNode)
-
-			nodeInfo, err := clusterSnapshot.GetNodeInfo(hintedNode)
-			if err != nil {
-				return "", err
-			}
-
-			if isNodeAcceptable(nodeInfo) {
-				return hintedNode, nil
-			}
+			return hintedNode, nil
+		} else if err != nil && err.Type() != clustersnapshot.FailingPredicateError {
+			// Unexpected error.
+			return "", err
 		}
 	}
 	return "", nil
 }
 
-func (s *HintingSimulator) findNode(similarPods *SimilarPodsScheduling, clusterSnapshot clustersnapshot.ClusterSnapshot, pod *apiv1.Pod, loggingQuota *klogx.Quota, isNodeAcceptable func(*framework.NodeInfo) bool) string {
+func (s *HintingSimulator) trySchedule(similarPods *SimilarPodsScheduling, clusterSnapshot clustersnapshot.ClusterSnapshot, pod *apiv1.Pod, loggingQuota *klogx.Quota, isNodeAcceptable func(*framework.NodeInfo) bool) (string, error) {
 	if similarPods.IsSimilarUnschedulable(pod) {
 		klogx.V(4).UpTo(loggingQuota).Infof("failed to find place for %s/%s based on similar pods scheduling", pod.Namespace, pod.Name)
-		return ""
+		return "", nil
 	}
 
-	newNodeName, err := s.predicateChecker.FitsAnyNodeMatching(clusterSnapshot, pod, isNodeAcceptable)
-	if err != nil {
+	newNodeName, err := clusterSnapshot.SchedulePodOnAnyNodeMatching(pod, isNodeAcceptable)
+	if err != nil && err.Type() == clustersnapshot.NoNodesPassingPredicatesFoundError {
 		klogx.V(4).UpTo(loggingQuota).Infof("failed to find place for %s/%s: %v", pod.Namespace, pod.Name, err)
 		similarPods.SetUnschedulable(pod)
-		return ""
+		return "", nil
+	} else if err != nil {
+		// Unexpected error.
```

Comment:
Do we want to classify `SchedulingInternalError`, `FailingPredicateError` as "Unexpected error" like this? The latter in particular seems like we might want to set to unschedulable like we're doing on L113 above.

Reply:
Yeah, this check is definitely missing `FailingPredicateError` - my bad, added.

Reply:
Changed this to check against `SchedulingInternalError` following Bartek's comment.